aptos-move/aptos-gas-meter/src/meter.rs

```diff
@@ -565,13 +565,22 @@ where
             .map_err(|e| e.finish(Location::Undefined))
     }
 
-    fn charge_keyless(&mut self) -> VMResult<()> {
+    fn charge_keyless(&mut self, num_authenticators: u8) -> VMResult<()> {
         if self.feature_version() < RELEASE_V1_12 {
             return Ok(());
         }
-
-        self.algebra
-            .charge_execution(KEYLESS_BASE_COST)
-            .map_err(|e| e.finish(Location::Undefined))
+        if self.feature_version() < RELEASE_V1_29 {
+            self.algebra
+                .charge_execution(KEYLESS_BASE_COST)
+                .map_err(|e| e.finish(Location::Undefined))
+        } else {
+            // Charge KEYLESS_BASE_COST for each authenticator
+            for _ in 0..num_authenticators {
+                self.algebra
+                    .charge_execution(KEYLESS_BASE_COST)
```

Comment:
I think you can do `num_authenticators * BASE_COST` (probably need to wrap number of authenticators in some gas param), callbacks are not cheap, if we can charge once, we should do that.